x/xvalidator/validator.go

```diff
@@ -17,7 +21,55 @@ const (
 	envSeparator  = "="
 )
 
-var envNameRegexp = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9_]*$")
+// New returns a new instance of 'validate' with more validation fields.
```

Comment:
Could you also move the function `validateServiceStruct ` from service validator package and make it generic:
https://github.com/mesg-foundation/engine/blob/6df2444c43929f697e5a41105a3d4cd404a2a2b2/service/validator/validator.go#L28-L45

to something like:
```go
func Validate(s interface{}) error {
	 validate, translator := New()
	var errs xerrors.Errors
	// validate service struct based on tag
	if err := validate.Struct(s); err != nil {
		for _, e := range err.(validator.ValidationErrors) {
			// Remove the name of the struct and the field from namespace
			trimmedNamespace := strings.TrimPrefix(e.Namespace(), namespacePrefix)
			trimmedNamespace = strings.TrimSuffix(trimmedNamespace, e.Field())
			// Only use it when in-cascade field
			namespace := ""
			if e.Field() != trimmedNamespace {
				namespace = trimmedNamespace
			}
			errs = append(errs, fmt.Errorf("%s%s", namespace, e.Translate(translator)))
		}
	}
	return errs.ErrorOrNil()
}
```

I guess having a simple helper that manage improve the error and manage the translation will be better ;)

Reply:
Also, the remaining functions in service/validator package should be move to service package.

Reply:
For now I left it in the service, because I'm not sure if we want such generic validation everywhere. It was designed for service only. 

But I've moved the creation of `Validate`